geth/api/backend_txqueue_test.go

```diff
@@ -391,28 +391,26 @@ func (s *BackendTestSuite) TestCompleteMultipleQueuedTransactions() {
 
 		// complete
 		results := s.backend.CompleteTransactions(string(updatedTxIDStrings), TestConfig.Account1.Password)
-		if len(results) != (testTxCount+1) || results["invalid-tx-id"].Error.Error() != "transaction hash not found" {
-			s.Fail(fmt.Sprintf("cannot complete txs: %v", results))
-			return
-		}
+		require.Len(results, testTxCount+1)
+		require.EqualError(results["invalid-tx-id"].Error, "transaction hash not found")
+
 		for txID, txResult := range results {
 			if txResult.Error != nil && txID != "invalid-tx-id" {
-				s.Fail(fmt.Sprintf("invalid error for %s", txID))
-				return
+				require.Fail(fmt.Sprintf("invalid error for %s", txID))
```

Comment:
In the spirit of this test refactoring, would be better to replace those `if`s to `requite.Equal`/`require.NoError`/etc as well.

Reply:
I've argued with myself about this because that assertion won't be really nice looking. Anyway, 2.5 against 0.5, I'll change that.